vtds_cluster_kvm/private/scripts/deploy_cluster_to_blade.py

```diff
@@ -1010,93 +398,30 @@ class VirtualNode:
     """A class for composing, creating and managing Virtual Nodes.
 
     """
-    def __init__(self, node_class, networks, node_instance):
+    def __init__(self, config, node_class, node_instance):
         """Constructor: 'node_class' is the node class configuration
         for the Virtual node, 'networks' is a filtered dictionary
         indexed by network name of the networks that are connected to
         the Virtual Node, 'node_instance' is the instance number
         within its node class of the Virtual Node.
 
         """
-        self.class_name = node_class['class_name']
         self.node_class = node_class
-        self.networks = networks
         self.node_instance = node_instance
-        self.hostname = self.__compute_hostname()
-        self.node_name = self.__compute_node_name()
-        self.nodeclass_dir = path_join(
-            os.sep, 'var', 'local', 'vtds', self.class_name
-        )
-        self.host_dir = path_join(self.nodeclass_dir, self.node_name)
-        makedirs(self.host_dir, mode=0o755, exist_ok=True)
-        self.boot_disk_name = None
-        try:
-            self.virtual_machine = self.node_class['virtual_machine']
-        except KeyError as err:
+        self.networks = node_connected_networks(node_class, config['networks'])
+        class_name = node_class['class_name']
+        if self.node_class.get('virtual_machine', None) is None:
             raise ContextualError(
                 "configuration error: node class '%s' does not define "
                 "a 'virtual_machine' section: %s" % (
-                    self.class_name, str(self.node_class)
-                )
-            ) from err
-        self.context = self.__compose()
-
```

Comment:
REVIEWERS: these methods were either turned into regular functions and moved into `cluster_common.py` or moved into the Node Builder or Disk Builder classes. You can find them there.